### ISSUE TITLE
Failed to download Metals. Not respecting the path provided in Java Home setting

### ISSUE THREAD

**Issue by tamipangadil** [scalameta/metals#2692]
**Describe the bug**

I used `asdf-vm` to manage my multiple Java versions. The Java Home path is not the same with regular Java installation. After setting the Java Home path in the settings, I'm not able to make the extension work in my VS Code which is strange. 

I get these error messages:
```
Java home: /home/USERNAME/.asdf/shims/
Metals version: 0.10.1
Downloading Metals failed with the following:
spawn /home/USERNAME/.asdf/shims/bin/java ENOENT
```

```
Failed to download Metals, make sure you have an internet connection and the Java Home '/home/USERNAME/.asdf/shims/' is valid. You can configure the Java Home in the settings.See https://scalameta.org/metals/docs/editors/vscode.html#http-proxy for instructions if you are using an HTTP proxy.
```

**To Reproduce**

_Download the Java JDK first using:_
1. Download Java using `asdf-vm`
2. Run these installation commands:
   ```
   asdf plugin add java
   asdf install java adoptopenjdk-11.0.9+101
   asdf global java adoptopenjdk-11.0.9+101
   ```

_Configure your VS Code_
1. Download the Scala (Metals) extension
2. Go to VS Code 'Settings'
2. Search for `@ext:scalameta.metals`
3. Scroll down to 'Metals: Java Home'
4. Set the input to `/home/{YOUR USER NAME HERE}/.asdf/shims/`
4. See error on the `OUTPUT` terminal tab

**Expected behavior**

It should able to know that the path being set in the Settings page is absolute and it should not look for extra prefix folder like `/bin`.

**Screenshots**
```
Java home: /home/USERNAME/.asdf/shims/
Metals version: 0.10.1
Downloading Metals failed with the following:
spawn /home/USERNAME/.asdf/shims/bin/java ENOENT
```

**Installation:**

- Operating system: Linux (Manjaro XFCE)
- Editor: Visual Studio Code
- Metals version: v0.10.2

**Additional context**

**Search terms**

**Comment by tgodzik** [comment]
Thanks for reporting, it seems that `/home/USERNAME/.asdf/shims/bin/java` doesn't exist. Can you confirm that it's available there?

**Comment by tamipangadil** [comment]
Yup, it doesn't exist. The `asdf-vm` tool doesn't install it similar to the regular Java installation.

**Comment by tgodzik** [comment]
That would be the problem then. Overall, `$JAVA_HOME/bin/java` is the standard place the executable should be. We don't have a way to use an absolute path, since the setting is `javaHome` and not `javaBinary`. We could try to add a fallback to check if what is being passed is actually an executable, but I estimate you would have different problems with not specifying an actual java_home (no access to JDK sources for example).

I will transfer the issue to `metals-vscode`, where to logic for this is, but I can't promise we will be able to take a look at it any time soon. We are happy to accept contributions, I don't estimate it would be hard to do.

**Comment by tgodzik** [comment]
Or is the actual path `/home/USERNAME/.asdf/shims/java` ?  Either way, I would recommend using a standard java_home, as I estimate Metals is not the only one that depends on the particular structure.
